lib/src/base-styles/_nx-colors.scss

```diff
@@ -6,7 +6,7 @@
  */
 :root {
   --nx-color-border: var(--nx-swatch-indigo-90);
-
+  --nx-color-border-secondary: var(--nx-swatch-indigo-95);
```

Comment:
Hmm should it be this or `--nx-color-secondary-border`? Is "secondary" more of an "element" or a "circumstance" (to use the terminology from the CSS variable docs page)? I especially ask in light of there being `--nx-color-form-element-border`

Reply:
> Hmm should it be this or `--nx-color-secondary-border`? Is "secondary" more of an "element" or a "circumstance" (to use the terminology from the CSS variable docs page)? I especially ask in light of there being `--nx-color-form-element-border`

IMO the form variables are incorrect. The variables names should be general to the left and more specific to the right.  The most general is that it's a color, next is that it's a border, next what kind of border, etc

Reply:
> The most general is that it's a color, next is that it's a border, next what kind of border, etc

I don't really agree.  Consider that some types of elements have variables for both border and background.  `color -> border|background -> element` isn't any more correctly sorted by than `color -> element -> border|background`.  They're both equally valid in terms of "general to the left and more specific to the right".  The latter is what we agreed on and wrote down though when we created the variable naming scheme, so it's what we should stick to.

Reply:
> > The most general is that it's a color, next is that it's a border, next what kind of border, etc
> 
> I don't really agree. Consider that some types of elements have variables for both border and background. `color -> border|background -> element` isn't any more correctly sorted by than `color -> element -> border|background`. They're both equally valid in terms of "general to the left and more specific to the right". The latter is what we agreed on and wrote down though when we created the variable naming scheme, so it's what we should stick to.

I think that you're correct in terms of borders/backgrounds that are a part of elements. Forms, lists, tables, etc. But in this case what you are suggesting is `color -> variation -> element`, because `secondary` isn't an element it's a variation of the element.

Reply:
> because secondary isn't an element it's a variation of the element.

Well that's what I was getting at when I said 

> Is "secondary" more of an "element" or a "circumstance" (to use the terminology from the CSS variable docs page)?

My thinking is that there aren't single elements that would have the primary border in some circumstances and a secondary border in other circumstances.  Rather, the elements that would get a secondary border are a completely different set than those that would get a primary border.  So to me secondary seems like an element as far as our naming scheme goes.  After all, the main purpose of this variable is to replace `--nx-color-subsection-border`, and `subsection` was definitely an element, so shouldn't this be too?

Reply:
> > because secondary isn't an element it's a variation of the element.
> 
> Well that's what I was getting at when I said
> 
> > Is "secondary" more of an "element" or a "circumstance" (to use the terminology from the CSS variable docs page)?
> 
> My thinking is that there aren't single elements that would have the primary border in some circumstances and a secondary border in other circumstances. Rather, the elements that would get a secondary border are a completely different set than those that would get a primary border. So to me secondary seems like an element as far as our naming scheme goes. After all, the main purpose of this variable is to replace `--nx-color-subsection-border`, and `subsection` was definitely an element, so shouldn't this be too?

We were also trying to get rid of the `subsection` border in general because it was being used by things that weren't subsections.

> My thinking is that there aren't single elements that would have the primary border in some circumstances and a secondary border in other circumstances

One example of such a thing is the list item borders where the first list item has a default border but all subsequent list items have a secondary border. Tables are similar where `thead` rows tend to be primary and `tbody` rows tend to be secondary.

When we talk about borders that are named specifically for the element they are attached to (form element, etc) it makes sense to name them `nx-color-form-element-border` because they will be in a set of other `nx-color-form-element-*` items (presumably) which makes them easy to group and easy to find. There isn't a set of `nx-color-secondary-*` elements, nor do I think that there should be. To me it makes more sense that the "basic" elements like `nx-color-border` and `nx-color-text` be named such that the variations are last, which if you look at `_nx-colors.scss` the majority are:

```
  --nx-color-interactive-background-selected: var(--nx-swatch-blue-90);
  --nx-color-interactive-background-active: var(--nx-swatch-blue-95);
  etc
  
  --nx-color-text-dark: var(--nx-swatch-grey-05);
  --nx-color-text: var(--nx-swatch-grey-10);
  --nx-color-text-disabled: var(--nx-color-interactive-border-disabled); // match disabled text and border
  etc
  
  --nx-color-link: var(--nx-swatch-blue-40);
  --nx-color-link-active: var(--nx-swatch-blue-60);
  etc

  --nx-color-threat-unspecified: var(--nx-swatch-indigo-95);
  --nx-color-threat-none: var(--nx-swatch-blue-70);
  etc
```